src/components/tabController/TabBar.tsx

```diff
@@ -128,364 +128,289 @@ interface Props extends TabControllerBarProps, BaseComponentInjectedProps, Forwa
+    // StyleProp<ViewStyle> TODO:
+    width: _indicatorWidth,
+    left: _indicatorOffset
+  };
 
-      this.tabBarItems = React.Children.map(this.children, (child: Partial<ChildProps>, index: number) => {
-        // @ts-ignore TODO: typescript - not sure if this can be easily solved
-        return React.cloneElement(child, {
-          labelColor,
-          selectedLabelColor,
-          labelStyle,
-          selectedLabelStyle,
-          uppercase,
-          iconColor,
-          selectedIconColor,
-          activeBackgroundColor,
-          ...child.props,
-          ...this.context,
-          index,
-          state: itemStates[index],
-          onLayout: this.onItemLayout
-        });
-      });
-      return this.tabBarItems;
-    }
-  }
+  const renderSelectedIndicator =
+    itemsWidths && itemsWidths.length > 0 ? (
+      <Reanimated.View style={[styles.selectedIndicator, indicatorStyle, _indicatorTransitionStyle]}/>
+    ) : undefined;
 
-  renderCodeBlock = _.memoize(() => {
-    const {currentPage, targetPage} = this.context;
-    const {itemsWidths, itemsOffsets} = this.state;
+  const renderCodeBlock = _.memoize(() => {
     const nodes: any[] = [];
 
-    nodes.push(set(this._indicatorOffset,
+    nodes.push(set(_indicatorOffset,
       interpolate(currentPage, {
         inputRange: itemsOffsets.map((_v, i) => i),
         outputRange: itemsOffsets
       })));
-    nodes.push(set(this._indicatorWidth,
+    nodes.push(set(_indicatorWidth,
       interpolate(currentPage, {inputRange: itemsWidths.map((_v, i) => i), outputRange: itemsWidths})));
 
-    nodes.push(Reanimated.onChange(targetPage, Reanimated.call([targetPage], this.focusSelected)));
+    nodes.push(Reanimated.onChange(targetPage, Reanimated.call([targetPage], focusSelected)));
 
     return <Code>{() => block(nodes)}</Code>;
   });
 
-  getShadowStyle() {
-    const {enableShadow, shadowStyle} = this.props;
-    return enableShadow && (shadowStyle || styles.containerShadow);
-  }
-
-  render() {
-    const {height, containerStyle, testID, backgroundColor} = this.props;
-    const {itemsWidths, scrollEnabled, fadeLeft, fadeRight} = this.state;
-    return (
-      <View style={[styles.container, this.getShadowStyle(), {width: this.containerWidth}, containerStyle]}>
-        <ScrollView
-          ref={this.tabBar}
-          horizontal
-          showsHorizontalScrollIndicator={false}
-          contentContainerStyle={{minWidth: this.containerWidth}}
-          scrollEnabled={scrollEnabled}
-          onScroll={this.onScroll}
-          scrollEventThrottle={16}
-          testID={testID}
-          snapToOffsets={this.getSnapBreakpoints()}
-          decelerationRate={'fast'}
+  const shadowStyle = enableShadow ? propsShadowStyle || styles.containerShadow : undefined;
+  return (
+    <View style={[styles.container, shadowStyle, {width: containerWidth}, containerStyle]}>
+      <FadedScrollView
+        /**
+         // @ts-ignore TODO: typescript */
```

Comment:
```suggestion
         // @ts-ignore TODO: typescript
```

Reply:
Weird, pretty sure it gave me an error previously...